builtin/providers/aws/resource_aws_dms_replication_subnet_group.go

```diff
@@ -0,0 +1,137 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	dms "github.com/aws/aws-sdk-go/service/databasemigrationservice"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDmsReplicationSubnetGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDmsReplicationSubnetGroupCreate,
+		Read:   resourceAwsDmsReplicationSubnetGroupRead,
+		Update: resourceAwsDmsReplicationSubnetGroupUpdate,
+		Delete: resourceAwsDmsReplicationSubnetGroupDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"replication_subnet_group_description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"replication_subnet_group_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateDmsReplicationSubnetGroupId,
+			},
+			"subnet_ids": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+				Required: true,
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsDmsReplicationSubnetGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dmsconn
+
+	// NOTE: Even though "tags" is an allowed parameter there is no way to retrieve the tags currently.
+	request := &dms.CreateReplicationSubnetGroupInput{
+		ReplicationSubnetGroupIdentifier:  aws.String(d.Get("replication_subnet_group_id").(string)),
+		ReplicationSubnetGroupDescription: aws.String(d.Get("replication_subnet_group_description").(string)),
+		SubnetIds:                         expandStringList(d.Get("subnet_ids").(*schema.Set).List()),
+	}
+
+	log.Println("[DEBUG] DMS create replication subnet group:", request)
+
+	response, err := conn.CreateReplicationSubnetGroup(request)
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsDmsReplicationSubnetGroupSetState(d, response.ReplicationSubnetGroup)
+}
+
+func resourceAwsDmsReplicationSubnetGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dmsconn
+
+	response, err := conn.DescribeReplicationSubnetGroups(&dms.DescribeReplicationSubnetGroupsInput{
+		Filters: []*dms.Filter{
+			{
+				Name:   aws.String("replication-subnet-group-id"),
+				Values: []*string{aws.String(d.Id())}, // Must use d.Id() to work with import.
+			},
+		},
+	})
+	if err != nil {
+		if dmserr, ok := err.(awserr.Error); ok && dmserr.Code() == "ResourceNotFoundFault" {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	return resourceAwsDmsReplicationSubnetGroupSetState(d, response.ReplicationSubnetGroups[0])
+}
+
+func resourceAwsDmsReplicationSubnetGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dmsconn
+
+	request := &dms.ModifyReplicationSubnetGroupInput{
```

Comment:
Do we have to pass everything as part of the Update request or should we pass only what we need to change?